pkg/agent/client_test.go

```diff
@@ -266,6 +266,9 @@ func TestFailedSend_DialResp_GRPC(t *testing.T) {
 	defer ts.Close()
 
 	func() {
+		// goleak.IgnoreCurrent() needs to include all the goroutines which were started by testClient.Serve() above.
+		// That includes things like pkg/agent/client.go:614 +0x88. Sleep does not guarantee they will have started.
+		time.Sleep(time.Second)
```

Comment:
Blech. I don't like this, but I don't have a simple alternative. I wonder if we could add pprof labeling to agent go routines, and then filter based on the pprof labels? Sounds cleaner than trying to parse stack traces, but not something I expect you to implement in this PR.

Reply:
Not a huge fan either. Right now I would like to get this race condition to mostly not happen. Its not representative of a real failure AFAICT. If the issue persists lets investigate a more complicated fix.